src/main/java/software/amazon/encryption/s3/internal/InstructionFileConfig.java

```diff
@@ -0,0 +1,115 @@
+package software.amazon.encryption.s3.internal;
+
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.core.async.AsyncResponseTransformer;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.encryption.s3.S3EncryptionClientException;
+
+/**
+ * Provides configuration options for instruction file behaviors.
+ */
+public class InstructionFileConfig {
+
+    final private InstructionFileClientType _clientType;
+    final private boolean _disableInstructionFile;
+    final private S3AsyncClient _s3AsyncClient;
+    final private S3Client _s3Client;
+
+    private InstructionFileConfig(final Builder builder) {
+        _clientType = builder._clientType;
+        _disableInstructionFile = builder._disableInstructionFile;
+        _s3Client = builder._s3Client;
+        _s3AsyncClient = builder._s3AsyncClient;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public enum InstructionFileClientType {
+        DISABLED,
+        SYNCHRONOUS,
+        ASYNC;
+    }
+
+    ResponseInputStream<GetObjectResponse> getInstructionFile(GetObjectRequest request) {
+        if (_disableInstructionFile) {
+            throw new S3EncryptionClientException("Instruction File has been disabled!");
+        }
+        switch (_clientType) {
+            case SYNCHRONOUS:
+                return _s3Client.getObject(request);
+            case ASYNC:
+                return _s3AsyncClient.getObject(request, AsyncResponseTransformer.toBlockingInputStream()).join();
+            case DISABLED:
+            default:
+                throw new S3EncryptionClientException("Unknown Instruction File Type");
```

Comment:
Oh... it looks like I screwed this up.
Disabled needs a different exception message.